collector/teradici_pcoip.go

```diff
@@ -0,0 +1,665 @@
+//go:build windows
+// +build windows
+
+package collector
+
+import (
+	"errors"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector("teradici_pcoip", NewTeradiciPcoipCollector)
+}
+
+// A TeradiciPcoipCollector is a Prometheus collector for WMI metrics:
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionAudioStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionGeneralStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionImagingStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionNetworkStatistics
+// Win32_PerfRawData_TeradiciPerf_PCoIPSessionUsbStatistics
+
+type TeradiciPcoipCollector struct {
```

Comment:
Types declared in these collectors should be private (lowercase first character) to avoid polluting the package API.
I 99% certain the other collectors don't do this, and I need to fix that at some stage.

Reply:
Thanks for catching this - should now be resolved